apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.V2Signer.cs

```diff
@@ -29,40 +29,42 @@ public sealed partial class UrlSigner
     {
         private sealed class V2Signer : ISigner
         {
-            public string Sign(
-                string bucket,
-                string objectName,
-                DateTimeOffset expiration,
-                HttpMethod requestMethod,
-                Dictionary<string, IEnumerable<string>> requestHeaders,
-                Dictionary<string, IEnumerable<string>> contentHeaders,
-                IBlobSigner blobSigner,
-                IClock clock)
+            public string Sign(RequestTemplate requestTemplate, Options options, IBlobSigner blobSigner, IClock clock)
             {
-                var state = new SigningState(bucket, objectName, expiration, requestMethod, requestHeaders, contentHeaders, blobSigner);
+                var state = new SigningState(
+                    requestTemplate.Bucket, 
+                    requestTemplate.ObjectName, 
+                    options.ToExpiration(clock).Expiration.Value,
+                    requestTemplate.HttpMethod,
+                    requestTemplate.RequestHeaders,
+                    requestTemplate.ContentHeaders,
+                    blobSigner);
                 var signature = blobSigner.CreateSignature(state._blobToSign);
                 return state.GetResult(signature);
             }
 
             public async Task<string> SignAsync(
-                string bucket,
-                string objectName,
-                DateTimeOffset expiration,
-                HttpMethod requestMethod,
-                Dictionary<string, IEnumerable<string>> requestHeaders,
-                Dictionary<string, IEnumerable<string>> contentHeaders,
+                RequestTemplate requestTemplate,
+                Options options,
                 IBlobSigner blobSigner,
                 IClock clock,
                 CancellationToken cancellationToken)
             {
-                var state = new SigningState(bucket, objectName, expiration, requestMethod, requestHeaders, contentHeaders, blobSigner);
+                var state = new SigningState(
```

Comment:
It may already be in a later commit, but I think it would make sense to have a SigningState constructor accepting a RequestTemplate.